arangod/Indexes/SortedIndexAttributeMatcher.cpp

```diff
@@ -333,38 +333,67 @@ Index::FilterCosts SortedIndexAttributeMatcher::supportsFilterCondition(
         // condition we have will make it even more selectivity. so in this case
         // we will re-use the selectivity estimate from the other index, and are
         // happy.
+        double otherEstimate = -1.0;
+  
         for (auto const& otherIdx : allIndexes) {
           auto const* other = otherIdx.get();
           if (other == idx || !other->hasSelectivityEstimate()) {
             continue;
           }
-          size_t matches = 0;
           auto const& otherFields = other->fields();
+          if (otherFields.size() > idx->fields().size()) {
+            // filter out too long other indexes
+            continue;
+          }
+          
+          size_t matches = 0;
           for (size_t i = 0; i < otherFields.size(); ++i) {
             if (otherFields[i] != idx->fields()[i]) {
               break;
             }
             ++matches;
-          }
-          if (matches == otherFields.size()) {
-            double estimate = other->selectivityEstimate();
-            if (estimate > 0.0) {
-              // reuse the estimate from the other index
-              estimatedItems = static_cast<double>(1.0 / estimate * values);
-            } else {
-              // use a guesstimate
-              estimatedItems /= equalityReductionFactor; 
+              
+            if (matches == otherFields.size()) {
+              // the other index is a full prefix of our own index.
```

Comment:
You have now moved this `if` into the loop, but does that make sense? The condition can only be fulfilled once we have iterated over all fields in `otherFields` (otherwise `matches` cannot equal `otherFields.size()`).